File: 25.12.25/string_to_integer.java
class Solution {
    public int myAtoi(String s) {
        s = s.trim();
        long tot=0;
        int i=0;
        int sign = 1;
        if(s.isEmpty()) return 0;
        if(s.charAt(i)=='+' || s.charAt(i)=='-'){
            if(s.charAt(i)=='-'){
                sign=-1;
                }
                i++;
        }
        for(int j=i;j<s.length();j++){
            char x = s.charAt(j);
            if(x>='0' &&  x<='9'){
                int num = x-'0';
                tot = (tot*10)+num;
                if(tot*sign >= Integer.MAX_VALUE) return Integer.MAX_VALUE;
                if(tot*sign <= Integer.MIN_VALUE) return Integer.MIN_VALUE;
            }
            else{
                break;
            }
            
        }return (int)tot*sign;
    }
}

File: 02.01.26/string_to_integer.py
class Solution(object):
    def myAtoi(self, s):
        """
        :type s: str
        :rtype: int
        """
        i = 0
        n = len(s)
        sign = 1
        num = 0

        while i < n and s[i] == ' ':
            i += 1

        if i < n and (s[i] == '+' or s[i] == '-'):
            if s[i] == '-':
                sign = -1
            i += 1

        while i < n and s[i].isdigit():
            digit = ord(s[i]) - ord('0')
            num = num * 10 + digit
            i += 1

        num = sign * num

        num = max(min(num, 2**31 - 1), -2 ** 31)

        return num
        
